File: server/src/models/AiChatMessage.ts
import mongoose, { Schema, Document, Types } from 'mongoose';

export type AiRole = 'user' | 'model' | 'function';

export interface IAiChatMessage extends Document {
    _id: Types.ObjectId;
    sessionId: Types.ObjectId;
    role: AiRole;
    content: string; // Storing as JSON string if function call/response
    functionCallName?: string;
    createdAt: Date;
    updatedAt: Date;
}

const AiChatMessageSchema = new Schema<IAiChatMessage>({
    sessionId: {
        type: Schema.Types.ObjectId,
        required: true,
        ref: 'AiChatSession',
        index: true
    },
    role: {
        type: String,
        enum: ['user', 'model', 'function'],
        required: true
    },
    content: {
        type: String,
        required: true
    },
    functionCallName: {
        type: String,
        default: null
    }
}, {
    timestamps: true,
    collection: 'ai_chat_messages'
});

export const AiChatMessage = mongoose.model<IAiChatMessage>('AiChatMessage', AiChatMessageSchema);

File: splitpal/lib/models/ai_chat_message.dart
class AiChatMessage {
  final String id;
  final String sessionId;
  final String role;
  final String content;
  final DateTime createdAt;

  AiChatMessage({
    required this.id,
    required this.sessionId,
    required this.role,
    required this.content,
    required this.createdAt,
  });

  factory AiChatMessage.fromJson(Map<String, dynamic> json) {
    return AiChatMessage(
      id: json['_id'] ?? '',
      sessionId: json['sessionId'] ?? '',
      role: json['role'] ?? 'user',
      content: json['content'] ?? '',
      createdAt: json['createdAt'] != null 
          ? DateTime.parse(json['createdAt']) 
          : DateTime.now(),
    );
  }
}
